src/extension/tsconfigHelper.ts

```diff
@@ -30,17 +31,20 @@ export class TsConfigHelper {
                 return fileSystem.readFile(tsConfigPath, "utf-8");
             })
             .then(function(jsonContents: string): Q.Promise<any> {
-                return JSON.parse(jsonContents);
+                return JSON.parse(stripJsonComments(jsonContents));
             });
     }
 
     /**
      * Writes out a JSON configuration object to the tsconfig.json file.
      */
     public static writeConfigJson(configJson: any): Q.Promise<void> {
+        return;
```

Comment:
Since we only use this to set two variables which only matter for non-typescript projects, I'm inclined to suggest the following: Replace this class with one that just has a `createDefaultTsConfigIfRequired` function which checks for the existence of a tsconfig.json file, and if one does not exist it creates it with `"allowjs": true` and also excluding node_modules. If something already exists, then do not touch it and assume that the user has configured it the way that they want.